### PR TITLE
TSP-01 & TSP-02 & TSP-03 fixes

### PR DESCRIPTION
**Pull request by dovgopoly**

### CODE REVIEW

File: contracts/gov/proposals/TokenSaleProposal.sol [hunk 1]
```diff
@@ -417,24 +433,31 @@ contract TokenSaleProposal is ITokenSaleProposal, ERC1155SupplyUpgradeable {
...
-        uint256 stepsCount = vestingSettings.vestingDuration / vestingSettings.unlockStep;
-        uint256 tokensPerStep = purchase.vestingTotalAmount / stepsCount;
+        if (timestamp >= purchase.purchaseTime + vestingSettings.vestingDuration) {
+            return purchase.vestingTotalAmount;
+        }
 
-        return
-            (uint256(vestingSettings.vestingDuration).min(timePoint - purchase.purchaseTime) /
-                vestingSettings.unlockStep) * tokensPerStep;
+        uint256 beforeSmallSegmentAmount = purchase.vestingTotalAmount.ratio(
```
Comment (Arvolear) [inline contracts/gov/proposals/TokenSaleProposal.sol:452]:
```suggestion
        uint256 beforeLastSegmentAmount = purchase.vestingTotalAmount.ratio(
```

File: contracts/gov/proposals/DistributionProposal.sol [hunk 2]
```diff
@@ -63,24 +65,25 @@ contract DistributionProposal is IDistributionProposal, Initializable {
             require(!dpInfo.claimed[voter], "DP: already claimed");
 
             uint256 reward = getPotentialReward(proposalIds[i], voter);
-            uint256 balance = address(rewardToken).thisBalance();
 
             dpInfo.claimed[voter] = true;
```
Comment (Arvolear) [inline contracts/gov/proposals/DistributionProposal.sol:70]: Can we refactor the code below like that? Using [token balance lib](https://github.com/dexe-network/investment-contracts/blob/master/contracts/libs/utils/TokenBalance.sol#L16).

```
rewardToken.sendFunds(voter, reward);

emit DistributionProposalClaimed(proposalIds[i], voter, reward);
```

Let's remove the logic with `safeTransferFrom()` as it is really complex and unobvious.

File: contracts/gov/proposals/TokenSaleProposal.sol (hunk 1 above)
Comment (dovgopoly) [reply contracts/gov/proposals/TokenSaleProposal.sol]: done

File: contracts/gov/proposals/DistributionProposal.sol (hunk 2 above)
Comment (dovgopoly) [reply contracts/gov/proposals/DistributionProposal.sol:70]: done

**Pull request merged**